### PR TITLE
fix local storage issue

### LINKED ISSUE #9115: After submitting a challenge, remove its code from browser's local storage
Our storage of incomplete code has caused a ton of confusion. Many campers switch browsers or devices, then visit a previously completed challenge only to think that "Free Code Camp has lost my code!"

The reason we store campers' code in a browser is to prevent them from losing this code if their browser crashes. 

- [ ] As a camper, once I submit a solution to a challenge, my stored solution should be removed from my browser's local storage so I next time I view this challenge.

- [ ] As a camper, if I return to a challenge and a partial solution for that challenge is available in my local storage, it should load my partial solution (as it currently does). I should see a flash message that says "We have loaded your partial solution below. Good luck finishing and submitting it."

### PR DESCRIPTION
**Pull request by DealPete**
#### Pre-Submission Checklist

- [X] Your pull request targets the `staging` branch of FreeCodeCamp.
- [X] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [X] You have only one commit (if not, [squash](https://github.com/FreeCodeCamp/FreeCodeCamp/wiki/git-rebase#squashing-multiple-commits-into-one) them into one commit).
- [X] All new and existing tests pass the command `npm run test-challenges`. Use `git commit --amend` to amend any fixes.
#### Type of Change

- [X] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)
#### Checklist:

- [X] Tested changes locally.
- [X] Closes currently open issue (replace XXXX with an issue no): Closes #9115
#### Description

Implemented the following two user stories:

As a camper, once I submit a solution to a challenge, my stored solution should be removed from my browser's local storage so I next time I view this challenge.

 As a camper, if I return to a challenge and a partial solution for that challenge is available in my local storage, it should load my partial solution (as it currently does). I should see a flash message that says "We have loaded your partial solution below. Good luck finishing and submitting it."

### CODE REVIEW

File: client/commonFramework/end.js [hunk 1]
```diff
@@ -112,6 +112,7 @@ $(document).ready(function() {
         common.displayTestResults(tests);
         if (solved) {
           common.showCompletion();
+					common.codeStorage.removeStoredValue(common.challengeName);
```
Comment (crashuniverse) [inline client/commonFramework/end.js]: It is too right to be cool. I was trying to be funny, I mean it should be indented to left 8/10 spaces. Same for line 56 maybe too?

Comment (DealPete) [inline client/commonFramework/end.js]: heh, looks like FreeCodeCamp uses spaces for indentation, whereas I use tabs. I'm surprised the linter doesn't catch that. I'll fix this and update my pull request.

File: client/commonFramework/end.js [hunk 2]
```diff
@@ -112,6 +112,7 @@ $(document).ready(function() {
         common.displayTestResults(tests);
         if (solved) {
           common.showCompletion();
+          common.codeStorage.removeStoredValue(common.challengeName);
```
Comment (crashuniverse) [inline client/commonFramework/end.js]: We can just use `challengeName`.

Comment (DealPete) [inline client/commonFramework/end.js]: You're right. Actually, the existing code used common.challengeName, and I just imitated that. The code in this file seems to inconsistently prepend "common." to variables that are already destructured in the const statement above. While we're looking at it I may as well go through and remove the "common." from variables that don't need it.

Comment (crashuniverse) [inline client/commonFramework/end.js]: With at least the new code we're adding we can spot such trivial things and make it a little better. About _

> from variables that don't need it

I  don't think refactoring the entire file would be needed or expected in the commit. Nevertheless, great work.

**Comment by raisedadead** [comment]
@crashuniverse : Thanks for the review.
@DealPete : This looks good and works, thanks a lot. And great job! 

But there is a migration issue that I am worried about, and am sure that we would get hit by a lot of camper complaints about this.

**Use case :**

1. I have a couple of challenges already in local storage say C1 and C2 and stored in the profile as well which I completed prior to this feature roll out.
2. I visit a challenge for the first time say C3, write some code, save it and move to next.
3. I do challenge C4, write some code (did not save it though) and visit C5 and complete and save it.
4. I visit C4, I'll be shown the notification.
5. I visit C1 or C2, I'll be shown the notification.

Now the potential issue here is, because the PR will reach in mid-production, Step 5 would create confusion. Campers would feel that their solution for C1 and C2 wasn't saved in the first place. But it actually happened before we rolled out the feature.

While visiting C4 is the good case where the notif is valid and should be shown.

I am guessing we need to device a way if the challenge is stored prior to the feature roll out clear it silently without the notification.

/cc @BerkeleyTrue and @QuincyLarson 

@DealPete Hit me up on the chat if you need further clarification of the potential issue that this may cause. Let's discuss a strategy before landing this in production.

**Comment by QuincyLarson** [comment]
@raisedadead Thanks for thinking of these corner cases. I think we should plan to ship these changes at the same time we ship the React changes. This way we can direct people to a notice with a full changelog and can explain this change there.

**Comment by raisedadead** [comment]
@QuincyLarson yeah, that would be great! 

And while we do that I am thinking we can add a check for the time stamp, so that the notification is shown only for challenges saved partially from the point this reaches production. 

> The lesser annoying notifications the lesser the distraction and better the UX.

@BerkeleyTrue can you guide on how we take this into react?

**Comment by BerkeleyTrue** [comment]
@raisedadead It would be faster to get this fixed and merged so that I can see the final result. Once that is in I can move it into redux. 

@QuincyLarson I don't agree that this needs to be merged in with the react stuff at the same time. Ideally we can release this the following week so that we can test this separately if it seems like there are edge cases that we need to cover before releasing to the general public.

**Comment by QuincyLarson** [comment]
@BerkeleyTrue whatever you feel comfortable doing. Shipping react is a much bigger priority than fixing this UX issue, and we want to ship React on-time and with as much QA behind it as possible.

**Comment by raisedadead** [comment]
Closed by https://github.com/FreeCodeCamp/FreeCodeCamp/pull/10200
